src/bentoml/_internal/frameworks/transformers.py

```diff
@@ -605,6 +607,266 @@ def make_default_signatures(pretrained: t.Any) -> ModelSignaturesType:
+
+        import bentoml
+
+        bentoml.transformers.import_model(
+            'my_t5_model',
+            "t5-base",
+            signatures={
+                "__call__": {"batchable": False},
+            }
+        )
+    """
+
+    tag = Tag.from_taglike(name)
+
+    if sync_with_hub_version:
+        if tag.version is not None:
+            logger.warn(
+                f"sync_with_hub_version is True, user provided version {tag.version} may be overridden by huggingface hub's commit hash"
+            )
+
+    # The below API are introduced since 4.18
+    if pkg_version_info("transformers")[:2] >= (4, 18):
+        from transformers.utils import is_flax_available
+        from transformers.utils import is_tf_available
+        from transformers.utils import is_torch_available
+    else:
+        from .utils.transformers import is_flax_available
+        from .utils.transformers import is_tf_available
+        from .utils.transformers import is_torch_available
+
+    framework_versions = {"transformers": get_pkg_version("transformers")}
+    if is_torch_available():
+        framework_versions["torch"] = get_pkg_version("torch")
+    if is_tf_available():
+        from .utils.tensorflow import get_tf_version
+
+        framework_versions["tensorflow"] = get_tf_version()
+    if is_flax_available():
+        framework_versions.update(
+            {
+                "flax": get_pkg_version("flax"),
+                "jax": get_pkg_version("jax"),
+                "jaxlib": get_pkg_version("jaxlib"),
+            }
+        )
+    context = ModelContext(
+        framework_name="transformers", framework_versions=framework_versions
+    )
+
+    config = transformers.AutoConfig.from_pretrained(
+        model_name_or_path,
+        proxies=proxies,
+        revision=revision,
+        force_download=force_download,
+        resume_download=resume_download,
+        **extra_hf_hub_kwargs,
+    )
+
+    model = None
+    if os.path.isdir(model_name_or_path):
+        src_dir = model_name_or_path
+        if sync_with_hub_version:
+            raise BentoMLException(
+                "Cannot sync version with huggingface hub when importing a local model"
+            )
+    else:
+        try:
+            is_huggingface_hub_available()
+            if clone_repository:
+                from huggingface_hub import snapshot_download
+
+                src_dir = snapshot_download(
+                    model_name_or_path,
+                    proxies=proxies,
+                    revision=revision,
+                    force_download=force_download,
+                    resume_download=resume_download,
+                    **extra_hf_hub_kwargs,
+                )
+            else:
+                from transformers.utils import cached_file
+
+                from .utils.transformers import init_empty_weights
+
+                with init_empty_weights():
+                    model = transformers.AutoModel.from_pretrained(
```

Comment:
Btw, this will still download the model to cache. So on BentoCloud, once the pod once initialised, it will redownload again at runtime. So if users use `bentoml.transformers.import_model` in the service code this might happen.

So tbh I'm not sure. Maybe we should have a check before on L700 if the tag is already exists then we return the Model ref immediately.

Reply:
afaik, from_pretrained should be smart enough to detect if a certain version of a certain model has already been downloaded. It will download if not available otherwise, it grabs directly from the cache. 

I'm not too knowledgeable about the workflow on BentoCloud, can you give me a bit more info on what you mean? What do you mean by 'redownload' once the pod has been initialised?

Reply:
I will always download the models if there is no cache. On bentocloud, we won't have the HUGGINGFACE_HUB_CACHE since the model is saved into bentoml models. So it will always download. Can you test it on bentocloud?

Reply:
After playing around with BentoCloud for a while, I have a few questions:

1. Would users call `bentoml.transformers.import_model` during runtime code? To my understanding, the normal workflow would be to save models to bento model store using `import_model` or `save_model`. Then we create a bento and push to bento cloud. I don't see the likelihood of users calling `import_model` during runtime?
2. Is this the change you have in mind:
```
tag = Tag.from_taglike(name)
try: 
    model = bentoml.models.get(tag)
    return model
except bentoml.exceptions.NotFound:
    pass
```

Reply:
An example service file
```python

runner = bentoml.transformers.import_model("llama-7b", "meta-llama/Llama-2-7b-chat-hf").to_runner()

svc = bentoml.Service('llama-chat', runners=[runner])

```

Reply:
I guess here `bentoml.transformers.import_model` is for importing models before packaging the bento and in service.py user should use `bentoml.transformers.get` instead of `bentoml.transformers.import_model` to get the bento Model. Does this work flow make sense for you?

Reply:
ye I guess that works. I guess this would be more structured once we have BentoProject. Since I feel like having just one file that has one line to imports the models is excessive. Since `bentoml.transformers.import_model` returns a `bentoml.Model` users might do this

I think for correctness I suggest `import_model` for both transformers and diffusers not to return anything. cc @larme ?

Reply:
> For correctness I suggest `import_model` for both transformers and diffusers not to return anything

Doesn't this break the backwards compatibility of users who already use diffusers import_model. 

e.g. if we make the above change, users who have done `diffusers.import_model().to_runner()` would now receive errors.

Reply:
I do agree with Aaron on the above point regarding some users may want their setup code and runtime code together in one service.py file (e.g doing `bentoml.transformers.import_model().to_runner()` during runtime). Though this is most likely a bad practice. I can still add a check on the tag and call bentoml.models.get on the tag at the beginning of the function.

Reply:
I think since we never document that behaviour, it is fine. @larme are you ok with breaking this?

Tho I do agree `import_model().to_runner()` is a very very bad practice.

Reply:
I guess `import_model` that return a bento model is helpful in notebook examples. For example:

```python
bento_model = import_model(...)

# then after several cells
runner = bento_model.with_options(...).to_runner()
```

I think just don't emphasize it in our documentation should be OK. So user don't have an understanding of the relationship bewteeen bento model and runner won't do that?

Reply:
Hmm, if we document in jupyter notebook then ppl will def put it into their service file.

Reply:
But I'm fine with this as well.